# **📊 1: Load spell data and show overview**

In [0]:
%sql
SELECT COUNT(DISTINCT *) AS total_spells FROM eldenringcatalog.gold.fact_spell_usage;

total_spells 213

In [0]:
%sql
SELECT 
    DISTINCT spell_name,
    spell_type,
    required_intelligence,
    required_faith,
    required_arcane,
    fp_cost_min,
    slots_required
FROM eldenringcatalog.gold.fact_spell_usage
LIMIT 10;

spell_name spell_type required_intelligence required_faith required_arcane fp_cost_min slots_required Flame Fortification Incantation 0 10 0 20 1 Blessing of the Erdtree Incantation 0 38 0 60 1 Elden Stars Incantation 0 50 0 41 2 Flame, Protect Me Incantation 0 24 0 30 1 The Flame of Frenzy Incantation 0 16 0 16 1 Dragonbolt Blessing Incantation 0 21 0 20 1 Heal from Afar Incantation 0 18 0 45 1 Lightning Fortification Incantation 0 10 0 20 1 Blessing's Boon Incantation 0 24 0 30 1 Flame Sling Incantation 0 10 0 11 1

# **📊 2: How many sorceries vs incantations?**

In [0]:
%sql
SELECT 
    spell_type,
    COUNT(DISTINCT *) AS spell_count
FROM eldenringcatalog.gold.fact_spell_usage
GROUP BY spell_type;

spell_type spell_count Incantation 129 Sorcery 84

# **📊 3: Analyze FP cost distribution**


**✔️ Simple Explanation**

**If a spell has:**

**fp_cost_min → the minimum FP it uses (usually a tap or quick cast)**

**fp_cost_max → the maximum FP it uses (usually a charged, held, or extended cast)**

In [0]:
%sql
SELECT 
    DISTINCT spell_name,
    spell_type,
    fp_cost_min,
    fp_cost_max,
    fp_cost_per_use
FROM eldenringcatalog.gold.fact_spell_usage
ORDER BY fp_cost_min DESC
LIMIT 30;

spell_name spell_type fp_cost_min fp_cost_max fp_cost_per_use Erdtree Heal Incantation 65 65 0 Placidusax's Ruin Incantation 62 62 0 Meteorite of Astel Sorcery 60 12 0 Blessing of the Erdtree Incantation 60 60 0 Ranni's Dark Moon Sorcery 57 57 0 Lord's Heal Incantation 55 55 0 Law of Regression Incantation 55 55 0 Greyoll's Roar Incantation 50 50 0 Rotten Butterflies Incantation 48 48 0 Borealis's Mist Incantation 48 6 0 Scarlet Aeonia Incantation 48 48 0 Light of Miquella Incantation 48 48 0 Ekzykes's Decay Incantation 48 6 0 Rellana's Twin Moons Sorcery 47 47 0 Golden Vow Incantation 47 47 0 Rennala's Full Moon Sorcery 47 47 0 Bayle's Tyranny Incantation 46 46 0 Theodorix's Magma Incantation 45 45 0 Great Heal Incantation 45 45 0 Heal from Afar Incantation 45 45 0 Bayle's Flame Lightning Incantation 43 43 0 Elden Stars Incantation 41 41 0 Mass of Putrescence Sorcery 41 41 0 Comet Azur Sorcery 40 10 0 Land of Shadow (Incantation) Incantation 40 40 0 Wrath of Gold Incantation 40 40 0 Loretta's Mastery Sorcery 39 39 0 Cannon of Haima Sorcery 38 38 0 Ghostflame Breath Incantation 36 36 0 Ancient Dragons' Lightning Strike Incantation 36 36 0

In [0]:
%sql
SELECT 
    AVG(fp_cost_min) AS avg_fp,
    MIN(fp_cost_min) AS min_fp,
    MAX(fp_cost_min) AS max_fp,
    STDDEV(fp_cost_min) AS stddev_fp
FROM eldenringcatalog.gold.fact_spell_usage;

avg_fp min_fp max_fp stddev_fp 23.671361502347416 1 65 12.440233591730212

**The analysis of spell FP consumption shows that the most demanding abilities are primarily high-tier incantations, with Erdtree Heal topping the list at 65 FP, followed by powerful boss-tier spells like Placidusax’s Ruin and Blessing of the Erdtree. Legendary sorceries such as Ranni’s Dark Moon and Rellana’s Twin Moons also appear among the highest-cost spells. Notably, several dragon incantations (Ekzykes’s Decay, Borealis’s Mist, Rotten Breath, Agheel’s Flame) have variable FP usage, allowing players to extend casting duration. When looking at overall spell data, spells average 23.67 FP, with a minimum of 1 FP, a maximum of 65 FP, and a standard deviation of 12.44, indicating a wide spread between basic utility spells and high-end, late-game abilities. Overall, FP consumption increases dramatically for advanced healing, buffs, dragon breaths, and legendary sorceries.**

# **📊 4: Most FP-efficient spells (lowest FP cost)**

In [0]:
%sql
SELECT 
    DISTINCT spell_name,
    spell_type,
    fp_cost_min,
    required_intelligence,
    required_faith,
    required_arcane
FROM eldenringcatalog.gold.fact_spell_usage
WHERE fp_cost_min > 0
ORDER BY fp_cost_min
LIMIT 20;

spell_name spell_type fp_cost_min required_intelligence required_faith required_arcane Surge, O Flame! Incantation 1 0 9 0 Divine Bird Feathers Incantation 3 0 24 0 Discus of Light Incantation 3 13 13 0 Carian Slicer Sorcery 4 14 0 0 Swift Glintstone Shard Sorcery 5 12 0 0 Briars of Sin Sorcery 6 0 24 0 Bestial Sling Incantation 7 0 10 0 Glintstone Pebble Sorcery 7 10 0 0 Thops's Barrier Sorcery 7 18 0 0 Cure Poison Incantation 7 0 10 0 Night Shard Sorcery 7 18 0 0 Carian Retaliation Sorcery 8 17 0 0 Glintstone Arc Sorcery 9 13 0 0 Rejection Incantation 9 0 12 0 Briars of Punishment Sorcery 9 0 21 0 Mantle of Thorns Sorcery 9 0 20 0 Starlight Sorcery 9 15 0 0 Miriam's Vanishing Sorcery 9 26 0 0 Bestial Constitution Incantation 10 0 9 0 Catch Flame Incantation 10 0 8 0

**the 20 lowest-cost spells in Elden Ring range from 1 to 10 FP, with many early-game friendly options. Incantations generally appear at the very bottom of the FP scale—like Surge, O Flame! at just 1 FP—while basic sorceries such as Carian Slicer and Glintstone Pebble remain efficient with requirements focused mainly on Intelligence. Most of these low-FP spells have minimal stat requirements, making them accessible to beginners and hybrid builds, and offering reliable, resource-efficient casting across both sorceries and incantations.**

# **📊 5: Most expensive spells (highest FP cost)**

In [0]:
%sql
SELECT 
    DISTINCT spell_name,
    spell_type,
    fp_cost_min,
    slots_required,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
ORDER BY fp_cost_min DESC
LIMIT 20;

spell_name spell_type fp_cost_min slots_required bonus Erdtree Heal Incantation 65 1 Erdtree Placidusax's Ruin Incantation 62 3 - Meteorite of Astel Sorcery 60 2 Gravity Blessing of the Erdtree Incantation 60 1 Erdtree Ranni's Dark Moon Sorcery 57 2 Full Moon Lord's Heal Incantation 55 1 Two Fingers Law of Regression Incantation 55 1 Golden Order Greyoll's Roar Incantation 50 2 Dragon Communion Light of Miquella Incantation 48 2 Miquellan Scarlet Aeonia Incantation 48 3 Servants of Rot Rotten Butterflies Incantation 48 1 Servants of Rot Borealis's Mist Incantation 48 1 Dragon Communion Ekzykes's Decay Incantation 48 1 Dragon Communion Golden Vow Incantation 47 1 Erdtree Rellana's Twin Moons Sorcery 47 2 Full Moon Rennala's Full Moon Sorcery 47 2 Full Moon Bayle's Tyranny Incantation 46 2 Dragon Communion Great Heal Incantation 45 1 Two Fingers Theodorix's Magma Incantation 45 1 Dragon Communion Heal from Afar Incantation 45 1 Two Fingers

**The spells with the highest FP costs—ranging from 45 to 65 FP—are powerful late-game sorceries and incantations tied to major factions like Erdtree, Dragon Communion, Full Moon, and Servants of Rot. These spells typically require 1–3 memory slots, reflecting their scale and impact. High-FP spells such as Erdtree Heal, Placidusax’s Ruin, and Meteorite of Astel deliver massive healing or devastating damage, making them resource-intensive but extremely potent tools meant for advanced characters and boss-level encounters.**

# **📊 6: Analyze spell slot requirements**

In [0]:
%sql
SELECT 
    slots_required,
    COUNT(DISTINCT *) AS spell_count
FROM eldenringcatalog.gold.fact_spell_usage
GROUP BY slots_required
ORDER BY slots_required;

slots_required spell_count 1 189 2 21 3 3

In [0]:
%sql
SELECT 
    DISTINCT spell_name,
    spell_type,
    slots_required,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE slots_required >= 2;

spell_name spell_type slots_required fp_cost_min bonus Giantsflame Take Thee Incantation 2 30 Fire Giant Messmer's Orb Incantation 2 31 Messmer Fire Bayle's Flame Lightning Incantation 2 43 Dragon Communion Placidusax's Ruin Incantation 3 62 - Scarlet Aeonia Incantation 3 48 Servants of Rot Minor Erdtree Incantation 2 30 Erdtree Greyoll's Roar Incantation 2 50 Dragon Communion Light of Miquella Incantation 2 48 Miquellan Elden Stars Incantation 2 41 Erdtree Black Blade Incantation 2 26 Erdtree Midra's Flame of Frenzy Incantation 2 22 Frenzied Flame Death Lightning Incantation 2 28 Dragon Cult Bayle's Tyranny Incantation 2 46 Dragon Communion Flame of the Fell God Incantation 2 34 Fire Giant Spira Incantation 2 10 Spiral Blades of Stone Sorcery 2 18 Gravity Vortex of Putrescence Sorcery 2 29 Death Founding Rain of Stars Sorcery 2 27 Primeval Sorceries Rykard's Rancor Sorcery 2 23 Magma Rennala's Full Moon Sorcery 2 47 Full Moon Meteorite of Astel Sorcery 2 60 Gravity Rellana's Twin Moons Sorcery 2 47 Full Moon Ranni's Dark Moon Sorcery 2 57 Full Moon Comet Azur Sorcery 3 40 Primeval Sorceries

**Multi-slot spells in Elden Ring are rare and represent some of the game’s most powerful and lore-significant abilities. All spells requiring 2 or more slots come from elite groups such as Dragon Communion, Erdtree, Full Moon, Gravity, Fire Giant, Magma, and Frenzied Flame, indicating strong thematic and gameplay importance. These spells also carry high FP costs, often between 40 and 60 FP, confirming their status as high-impact, late-game abilities. Only three spells—Placidusax’s Ruin, Scarlet Aeonia, and Comet Azur—require 3 slots, marking them as some of the strongest spells in the entire catalog.**

# **📊 7: Intelligence requirement distribution**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_intelligence,
    fp_cost_min
FROM eldenringcatalog.gold.fact_spell_usage
WHERE required_intelligence > 0
ORDER BY required_intelligence DESC
LIMIT 30;

spell_name spell_type required_intelligence fp_cost_min Rellana's Twin Moons Sorcery 72 47 Rennala's Full Moon Sorcery 70 47 Ranni's Dark Moon Sorcery 68 57 Comet Azur Sorcery 60 40 Meteorite of Astel Sorcery 55 60 Founding Rain of Stars Sorcery 52 27 Comet Sorcery 52 24 Blades of Stone Sorcery 48 18 Crystal Torrent Sorcery 47 20 Loretta's Mastery Sorcery 46 39 Stars of Ruin Sorcery 43 32 Explosive Ghostflame Sorcery 42 29 Fleeting Microcosm Sorcery 42 26 Crystal Release Sorcery 41 34 Rykard's Rancor Sorcery 40 23 Night Comet Sorcery 38 21 Shattering Crystal Sorcery 38 21 Law of Regression Incantation 37 55 Gravitational Missile Sorcery 36 18 Zamor Ice Storm Sorcery 36 17 Collapsing Stars Sorcery 36 18 Cherishing Fingers Sorcery 36 20 Glintstone Cometshard Sorcery 36 17 Eternal Darkness Sorcery 35 25 Ancient Death Rancor Sorcery 34 21 Carian Phalanx Sorcery 34 24 Glintstone Nails Sorcery 32 23 Adula's Moonblade Sorcery 32 22 Vortex of Putrescence Sorcery 32 29 Radagon's Rings of Light Incantation 31 21

In [0]:
%sql
SELECT 
    AVG(required_intelligence) AS avg_int_requirement
FROM eldenringcatalog.gold.fact_spell_usage
WHERE spell_type = 'Sorcery' AND required_intelligence > 0;

avg_int_requirement 28.225

**High-intelligence sorceries in Elden Ring represent the pinnacle of Glintstone magic, with requirements reaching up to 72 INT for Rellana’s Twin Moons, followed closely by Rennala’s Full Moon (70 INT) and Ranni’s Dark Moon (68 INT). These elite spells also come with very high FP costs, highlighting their immense late-game power. The dataset shows that top-tier sorceries consistently require 52+ INT, while the average intelligence requirement across all sorceries is 28 INT, meaning these high-end spells sit far above the norm. Overall, the findings confirm that Full Moon, Gravity, Primeval, and high-tier projectile sorceries dominate the upper range of intelligence scaling**

# **📊 8: Faith requirement distribution**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_faith,
    fp_cost_min
FROM eldenringcatalog.gold.fact_spell_usage
WHERE required_faith > 0
ORDER BY required_faith DESC
LIMIT 30;

spell_name spell_type required_faith fp_cost_min Light of Miquella Incantation 72 48 Minor Erdtree Incantation 70 30 Messmer's Orb Incantation 60 31 Land of Shadow (Incantation) Incantation 58 40 Rain of Fire Incantation 52 27 Dragonbolt of Florissax Incantation 52 35 Elden Stars Incantation 50 41 Spira Incantation 48 10 Death Lightning Incantation 47 28 Black Blade Incantation 46 26 Fortissax's Lightning Spear Incantation 46 35 Black Flame Ritual Incantation 42 30 Erdtree Heal Incantation 42 65 Flame of the Fell God Incantation 41 34 Midra's Flame of Frenzy Incantation 41 22 Lansseax's Glaive Incantation 40 22 Blessing of the Erdtree Incantation 38 60 Placidusax's Ruin Incantation 36 62 Multilayered Ring of Light Incantation 36 23 Knight's Lightning Spear Incantation 36 29 Protection of the Erdtree Incantation 35 30 Scarlet Aeonia Incantation 35 48 Wrath from Afar Incantation 34 18 Frozen Lightning Spear Incantation 34 29 Giant Golden Arc Incantation 34 24 Rotten Butterflies Incantation 33 48 Howl of Shabriri Incantation 33 21 Wrath of Gold Incantation 32 40 Ancient Dragons' Lightning Spear Incantation 32 25 Unendurable Frenzy Incantation 31 22

In [0]:
%sql
SELECT 
    AVG(required_faith) AS avg_faith_requirement
FROM eldenringcatalog.gold.fact_spell_usage
WHERE spell_type = 'Incantation' AND required_faith > 0;

avg_faith_requirement 24.312

**Faith-based spells in Elden Ring are dominated by Incantations, with the top 30 requiring 31 to 72 Faith. The spell with the highest Faith requirement is Light of Miquella (72 Faith, 48 FP), followed closely by Minor Erdtree (70 Faith, 30 FP). These spells include powerful healing, offensive, and utility effects, such as Erdtree Heal, Blessing of the Erdtree, and Midra's Flame of Frenzy. The average Faith requirement across all Faith-based Incantations is approximately 24.3, suggesting that while most spells are accessible to moderately Faith-focused characters, the strongest abilities demand significant investment in the Faith stat.**

# **📊 9: Arcane-based spells**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_arcane,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE required_arcane > 0
ORDER BY required_arcane DESC;

spell_name spell_type required_arcane fp_cost_min bonus Bayle's Flame Lightning Incantation 53 43 Dragon Communion Bayle's Tyranny Incantation 49 46 Dragon Communion Furious Blade of Ansbach Incantation 27 18 Blood Great Oracular Bubble Sorcery 18 16 Claymen Sorceries Bloodboon Incantation 17 13 Blood Greyoll's Roar Incantation 17 50 Dragon Communion Dragonmaw Incantation 16 34 Dragon Communion Swarm of Flies Incantation 16 14 Blood Bloodflame Talons Incantation 15 12 Blood Ghostflame Breath Incantation 15 36 Dragon Communion Smarag's Glintstone Breath Incantation 15 36 Dragon Communion Borealis's Mist Incantation 15 48 Dragon Communion Agheel's Flame Incantation 15 36 Dragon Communion Ekzykes's Decay Incantation 15 48 Dragon Communion Oracle Bubbles Sorcery 15 12 Claymen Sorceries Theodorix's Magma Incantation 14 45 Dragon Communion Dragonclaw Incantation 13 24 Dragon Communion Glintstone Breath Incantation 12 28 Dragon Communion Dragonice Incantation 12 36 Dragon Communion Rotten Breath Incantation 12 36 Dragon Communion Dragonfire Incantation 12 28 Dragon Communion Magma Breath Incantation 10 30 Dragon Communion Bloodflame Blade Incantation 10 20 Blood

**Arcane-based spells in Elden Ring are mostly Incantations tied to Dragon Communion or Blood bonuses. The highest Arcane requirements are Bayle's Flame Lightning (53) and Bayle's Tyranny (49), while moderate-cost spells like Greyoll's Roar and Borealis's Mist balance power and accessibility. Investing in Arcane unlocks some of the most unique and potent spells, offering strong utility and damage for specialized builds.**

# **📊 10: Spells requiring multiple stats**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_intelligence,
    required_faith,
    required_arcane,
    (CASE WHEN required_intelligence > 0 THEN 1 ELSE 0 END +
     CASE WHEN required_faith > 0 THEN 1 ELSE 0 END +
     CASE WHEN required_arcane > 0 THEN 1 ELSE 0 END) AS stat_count
FROM eldenringcatalog.gold.fact_spell_usage
WHERE (CASE WHEN required_intelligence > 0 THEN 1 ELSE 0 END +
       CASE WHEN required_faith > 0 THEN 1 ELSE 0 END +
       CASE WHEN required_arcane > 0 THEN 1 ELSE 0 END) >= 2
ORDER BY stat_count DESC;

spell_name spell_type required_intelligence required_faith required_arcane stat_count Magma Breath Incantation 0 14 10 2 Glintstone Breath Incantation 0 15 12 2 Smarag's Glintstone Breath Incantation 0 23 15 2 Radagon's Rings of Light Incantation 31 31 0 2 Litany of Proper Death Incantation 17 17 0 2 Bloodboon Incantation 0 14 17 2 Order Healing Incantation 11 11 0 2 Rotten Breath Incantation 0 15 12 2 Dragonice Incantation 0 15 12 2 Triple Rings of Light Incantation 23 23 0 2 Agheel's Flame Incantation 0 23 15 2 Immutable Shield Incantation 19 19 0 2 Ekzykes's Decay Incantation 0 23 15 2 Swarm of Flies Incantation 0 11 16 2 Furious Blade of Ansbach Incantation 0 19 27 2 Bloodflame Talons Incantation 0 13 15 2 Dragonfire Incantation 0 15 12 2 Order's Blade Incantation 13 13 0 2 Bloodflame Blade Incantation 0 12 10 2 Dragonmaw Incantation 0 24 16 2 Dragonclaw Incantation 0 17 13 2 Ghostflame Breath Incantation 0 23 15 2 Theodorix's Magma Incantation 0 21 14 2 Borealis's Mist Incantation 0 23 15 2 Greyoll's Roar Incantation 0 28 17 2 Discus of Light Incantation 13 13 0 2 Explosive Ghostflame Sorcery 42 30 0 2 Ancient Death Rancor Sorcery 34 24 0 2 Gelmir's Fury Sorcery 28 15 0 2 Roiling Magma Sorcery 21 12 0 2 Rings of Spectral Light Sorcery 24 18 0 2 Oracle Bubbles Sorcery 19 0 15 2 Fia's Mist Sorcery 23 18 0 2 Magma Shot Sorcery 19 10 0 2 Mass of Putrescence Sorcery 28 22 0 2 Vortex of Putrescence Sorcery 32 26 0 2 Rancorcall Sorcery 16 14 0 2 Great Oracular Bubble Sorcery 25 0 18 2 Tibia's Summons Sorcery 28 20 0 2 Rykard's Rancor Sorcery 40 18 0 2

**Many Elden Ring spells rely on multiple stats rather than a single attribute. These dual-stat spells, requiring combinations of Intelligence, Faith, and Arcane, provide versatile effects. For example, Radagon's Rings of Light uses both Intelligence and Faith, while Arcane-heavy Incantations like Glintstone Breath pair Arcane with Faith. Sorceries such as Explosive Ghostflame combine Intelligence and Faith for powerful damage. Investing in two stats can unlock these potent hybrid spells, offering flexibility and stronger scaling for builds that aren’t purely single-stat focused.**

# **📊 11: Calculate total stat requirements per spell**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_intelligence,
    required_faith,
    required_arcane,
    (required_intelligence + required_faith + required_arcane) AS total_stat_requirement,
    fp_cost_min
FROM eldenringcatalog.gold.fact_spell_usage
ORDER BY total_stat_requirement DESC
LIMIT 30;

spell_name spell_type required_intelligence required_faith required_arcane total_stat_requirement fp_cost_min Light of Miquella Incantation 0 72 0 72 48 Explosive Ghostflame Sorcery 42 30 0 72 29 Rellana's Twin Moons Sorcery 72 0 0 72 47 Rennala's Full Moon Sorcery 70 0 0 70 47 Minor Erdtree Incantation 0 70 0 70 30 Ranni's Dark Moon Sorcery 68 0 0 68 57 Radagon's Rings of Light Incantation 31 31 0 62 21 Comet Azur Sorcery 60 0 0 60 40 Messmer's Orb Incantation 0 60 0 60 31 Ancient Death Rancor Sorcery 34 24 0 58 21 Vortex of Putrescence Sorcery 32 26 0 58 29 Land of Shadow (Incantation) Incantation 0 58 0 58 40 Rykard's Rancor Sorcery 40 18 0 58 23 Meteorite of Astel Sorcery 55 0 0 55 60 Bayle's Flame Lightning Incantation 0 0 53 53 43 Dragonbolt of Florissax Incantation 0 52 0 52 35 Comet Sorcery 52 0 0 52 24 Founding Rain of Stars Sorcery 52 0 0 52 27 Rain of Fire Incantation 0 52 0 52 27 Mass of Putrescence Sorcery 28 22 0 50 41 Elden Stars Incantation 0 50 0 50 41 Bayle's Tyranny Incantation 0 0 49 49 46 Tibia's Summons Sorcery 28 20 0 48 17 Spira Incantation 0 48 0 48 10 Blades of Stone Sorcery 48 0 0 48 18 Death Lightning Incantation 0 47 0 47 28 Crystal Torrent Sorcery 47 0 0 47 20 Furious Blade of Ansbach Incantation 0 19 27 46 18 Black Blade Incantation 0 46 0 46 26 Loretta's Mastery Sorcery 46 0 0 46 39

**The spells with the highest total stat requirements in Elden Ring demand significant investment in one or more attributes. For instance, Light of Miquella and Explosive Ghostflame each require a combined 72 points across Intelligence and Faith, while classic Sorceries like Rellana's Twin Moons and Rennala's Full Moon demand high Intelligence alone. Hybrid spells like Radagon's Rings of Light use both Intelligence and Faith, offering flexibility but still requiring careful stat allocation. These high-requirement spells often have powerful effects or unique mechanics, making them highly rewarding for players willing to invest in the necessary stats.**

# **📊 12: FP efficiency metric (FP cost per stat requirement)**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    fp_cost_min,
    (required_intelligence + required_faith + required_arcane) AS total_reqs,
    ROUND(fp_cost_min / (required_intelligence + required_faith + required_arcane), 2) AS fp_per_stat
FROM eldenringcatalog.gold.fact_spell_usage
WHERE (required_intelligence + required_faith + required_arcane) > 0 
  AND fp_cost_min > 0
ORDER BY fp_per_stat
LIMIT 30;

spell_name spell_type fp_cost_min total_reqs fp_per_stat Surge, O Flame! Incantation 1 9 0.11 Discus of Light Incantation 3 26 0.12 Divine Bird Feathers Incantation 3 24 0.13 Spira Incantation 10 48 0.21 Briars of Sin Sorcery 6 24 0.25 Carian Slicer Sorcery 4 14 0.29 Rings of Spectral Light Sorcery 14 42 0.33 Radagon's Rings of Light Incantation 21 62 0.34 Tibia's Summons Sorcery 17 48 0.35 Oracle Bubbles Sorcery 12 34 0.35 Miriam's Vanishing Sorcery 9 26 0.35 Ancient Death Rancor Sorcery 21 58 0.36 Great Oracular Bubble Sorcery 16 43 0.37 Gelmir's Fury Sorcery 16 43 0.37 Blades of Stone Sorcery 18 48 0.38 Immutable Shield Incantation 15 38 0.39 Furious Blade of Ansbach Incantation 18 46 0.39 Thops's Barrier Sorcery 7 18 0.39 Night Shard Sorcery 7 18 0.39 Rykard's Rancor Sorcery 23 58 0.4 Rancorcall Sorcery 12 30 0.4 Explosive Ghostflame Sorcery 29 72 0.4 Bloodboon Incantation 13 31 0.42 Swift Glintstone Shard Sorcery 5 12 0.42 Crystal Torrent Sorcery 20 47 0.43 Minor Erdtree Incantation 30 70 0.43 Briars of Punishment Sorcery 9 21 0.43 Bloodflame Talons Incantation 12 28 0.43 Mantle of Thorns Sorcery 9 20 0.45 Watchful Spirit Incantation 12 26 0.46

**the most FP-efficient spells in Elden Ring, showing which spells consume the least FP relative to their total stat requirements. Spells like Surge, O Flame! and Discus of Light have extremely low FP-per-stat ratios (0.11–0.13), making them very cost-effective for players who want high utility without burning through FP. On the other hand, high-powered spells like Radagon's Rings of Light or Explosive Ghostflame still maintain reasonable FP efficiency (0.34–0.40) considering their high total stat requirements. This kind of analysis helps players optimize their builds, balancing spell power with sustainability in combat.**.

# **📊 13: Analyze spell "bonus" types (spell schools)**

In [0]:
%sql
SELECT 
    bonus,
    COUNT(distinct *) AS spell_count,
    ROUND(AVG(fp_cost_min), 2) AS avg_fp_cost
FROM eldenringcatalog.gold.fact_spell_usage
GROUP BY bonus
ORDER BY spell_count DESC;

bonus spell_count avg_fp_cost Glintstone 18 16.78 Two Fingers 16 24.06 Dragon Communion 16 37.75 Dragon Cult 15 26.4 Erdtree 15 34.93 Carian 12 15.08 Golden Order 9 21.44 Fire Monk 8 16.13 Death 8 23.5 Night 7 17.0 Gravity 7 24.86 Bestial 7 14.0 Frenzied Flame 6 22.83 Servants of Rot 6 29.33 Godskin Apostle 5 22.6 Blood 5 15.4 Cold 5 18.2 Fire Giant 4 26.5 Magma 4 20.75 Finger 4 19.75 Spiral 4 14.5 Messmer Fire 3 23.0 Crystalian 3 25.0 Full Moon 3 50.33 Primeval Sorceries 3 33.0 Erdtree [1][2] 3 22.0 Miquellan 2 35.5 Loretta's Sorceries 2 31.5 Claymen Sorceries 2 14.0 - 2 44.0 Aberrant 2 7.5 Scadutree 2 12.0 -- 2 13.5 Glintstone (Stonedigger) 1 10.0 Godslayer 1 21.0 Glintstone (Stonediggers) 1 22.0

**The data shows that Glintstone spells are the most common and FP-efficient, while Dragon and Erdtree spells are fewer but cost more FP, reflecting their power. Rare bonuses like Full Moon or Miquellan have very high FP costs, making them strong but specialized. Overall, bonus type strongly influences both spell frequency and FP efficiency.**

# **📊 14: Dragon spells**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_arcane,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE bonus LIKE '%Dragon%'
ORDER BY fp_cost_min DESC;

spell_name spell_type required_arcane fp_cost_min bonus Greyoll's Roar Incantation 17 50 Dragon Communion Borealis's Mist Incantation 15 48 Dragon Communion Ekzykes's Decay Incantation 15 48 Dragon Communion Bayle's Tyranny Incantation 49 46 Dragon Communion Theodorix's Magma Incantation 14 45 Dragon Communion Bayle's Flame Lightning Incantation 53 43 Dragon Communion Ghostflame Breath Incantation 15 36 Dragon Communion Smarag's Glintstone Breath Incantation 15 36 Dragon Communion Dragonice Incantation 12 36 Dragon Communion Rotten Breath Incantation 12 36 Dragon Communion Agheel's Flame Incantation 15 36 Dragon Communion Ancient Dragons' Lightning Strike Incantation 0 36 Dragon Cult Dragonbolt of Florissax Incantation 0 35 Dragon Cult Fortissax's Lightning Spear Incantation 0 35 Dragon Cult Vyke's Dragonbolt Incantation 0 35 Dragon Cult Dragonmaw Incantation 16 34 Dragon Communion Magma Breath Incantation 10 30 Dragon Communion Frozen Lightning Spear Incantation 0 29 Dragon Cult Knight's Lightning Spear Incantation 0 29 Dragon Cult Glintstone Breath Incantation 12 28 Dragon Communion Dragonfire Incantation 12 28 Dragon Communion Death Lightning Incantation 0 28 Dragon Cult Electrify Armament Incantation 0 27 Dragon Cult Electrocharge Incantation 0 26 Dragon Cult Ancient Dragons' Lightning Spear Incantation 0 25 Dragon Cult Dragonclaw Incantation 13 24 Dragon Communion Lansseax's Glaive Incantation 0 22 Dragon Cult Dragonbolt Blessing Incantation 0 20 Dragon Cult Lightning Strike Incantation 0 19 Dragon Cult Lightning Spear Incantation 0 18 Dragon Cult Honed Bolt Incantation 0 12 Dragon Cult

**The query highlights Dragon-themed spells, primarily from the Dragon Communion and Dragon Cult bonuses. These spells generally have high FP costs and often require Arcane, reflecting their potency and specialization. Dragon Communion spells tend to demand more Arcane points, while Dragon Cult spells usually rely less on stats but still have significant FP costs, emphasizing raw magical power over stat requirements.**

# **📊 15: Erdtree spells**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_faith,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE bonus LIKE '%Erdtree%'
ORDER BY required_faith DESC;

spell_name spell_type required_faith fp_cost_min bonus Minor Erdtree Incantation 70 30 Erdtree Land of Shadow (Incantation) Incantation 58 40 Erdtree Elden Stars Incantation 50 41 Erdtree Black Blade Incantation 46 26 Erdtree Erdtree Heal Incantation 42 65 Erdtree Blessing of the Erdtree Incantation 38 60 Erdtree Protection of the Erdtree Incantation 35 30 Erdtree Wrath from Afar Incantation 34 18 Erdtree Wrath of Gold Incantation 32 40 Erdtree Aspects of the Crucible: Thorns Incantation 27 14 Erdtree Aspects of the Crucible: Bloom Incantation 27 23 Erdtree Aspects of the Crucible: Breath Incantation 27 28 Erdtree [1][2] Aspects of the Crucible: Tail Incantation 27 20 Erdtree [1][2] Aspects of the Crucible: Horns Incantation 27 18 Erdtree [1][2] Golden Vow Incantation 25 47 Erdtree Blessing's Boon Incantation 24 30 Erdtree Barrier of Gold Incantation 24 30 Erdtree Golden Lightning Fortification Incantation 24 30 Erdtree

**The query shows Erdtree-themed incantations, which heavily emphasize Faith requirements. These spells tend to have moderate to high FP costs, reflecting their strong supportive or offensive abilities. The Erdtree bonus group focuses on boosting Faith-based spells, often used for healing, protection, or holy damage, making them ideal for Faith-focused builds.**

# **📊 16: Frenzied Flame spells**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_faith,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE bonus LIKE '%Frenzied%' 

spell_name spell_type required_faith fp_cost_min bonus Inescapable Frenzy Incantation 21 32 Frenzied Flame Howl of Shabriri Incantation 33 21 Frenzied Flame Unendurable Frenzy Incantation 31 22 Frenzied Flame Frenzied Burst Incantation 22 24 Frenzied Flame Midra's Flame of Frenzy Incantation 41 22 Frenzied Flame The Flame of Frenzy Incantation 16 16 Frenzied Flame

**all Frenzied Flame incantations, which are Faith-based spells known for their chaotic and aggressive effects. They generally require moderate Faith values (16–41) and have variable FP costs (16–32), reflecting a balance between accessibility and potency. These spells are ideal for players building around high-risk, high-reward Faith strategies.**

# **📊 17: DLC spells vs base game**

In [0]:
%sql
SELECT 
    CASE WHEN is_dlc = 1 THEN 'DLC' ELSE 'Base Game' END AS content_type,
    COUNT(DISTINCT *) AS spell_count,
    ROUND(AVG(fp_cost_min), 2) AS avg_fp_cost,
    ROUND(AVG(required_intelligence), 2) AS avg_int,
    ROUND(AVG(required_faith), 2) AS avg_fai
FROM eldenringcatalog.gold.fact_spell_usage
GROUP BY is_dlc;

content_type spell_count avg_fp_cost avg_int avg_fai DLC 42 24.79 10.05 24.67 Base Game 171 23.4 11.87 13.56

**This analysis compares spells from the **Base Game** versus **DLC content** in *Elden Ring*. DLC spells tend to have slightly **higher average FP costs** (24.79 vs. 23.4) and are **Faith-focused** (avg Faith 24.67 vs. 13.56), whereas Base Game spells have a slightly higher **Intelligence requirement** (avg 11.87 vs. 10.05). Overall, DLC content introduces spells that are more potent for Faith-based builds, offering specialized options beyond the Base Game.**


# **📊 18: Lowest stat requirement spells (beginner-friendly)**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_intelligence,
    required_faith,
    required_arcane,
    (required_intelligence + required_faith + required_arcane) AS total_reqs,
    fp_cost_min
FROM eldenringcatalog.gold.fact_spell_usage
WHERE (required_intelligence + required_faith + required_arcane) > 0
ORDER BY total_reqs
LIMIT 20;

spell_name spell_type required_intelligence required_faith required_arcane total_reqs fp_cost_min Beast Claw Incantation 0 8 0 8 10 Catch Flame Incantation 0 8 0 8 10 Urgent Heal Incantation 0 8 0 8 16 Bestial Constitution Incantation 0 9 0 9 10 Surge, O Flame! Incantation 0 9 0 9 1 Bestial Sling Incantation 0 10 0 10 7 Magic Fortification Incantation 0 10 0 10 20 Flame Sling Incantation 0 10 0 10 11 Glintstone Pebble Sorcery 10 0 0 10 7 Cure Poison Incantation 0 10 0 10 7 Assassin's Approach Incantation 0 10 0 10 15 Flame Fortification Incantation 0 10 0 10 20 Lightning Fortification Incantation 0 10 0 10 20 Divine Fortification Incantation 0 10 0 10 20 Poison Armament Incantation 0 10 0 10 15 Pest Threads Incantation 0 11 0 11 19 Poison Mist Incantation 0 12 0 12 18 Bestial Vitality Incantation 0 12 0 12 18 Lord's Aid Incantation 0 12 0 12 12 Flame, Cleanse Me Incantation 0 12 0 12 14

**The spells with the lowest total stat requirements in Elden Ring are mostly Faith-based Incantations, requiring between 8 and 12 total points. They generally have low FP costs, with notable exceptions like Urgent Heal (16 FP) and various Fortification spells (20 FP). A few early Sorceries, like Glintstone Pebble, also appear but focus on Intelligence instead of Faith. These spells are ideal for low-level or hybrid builds, providing utility without demanding high stat investment.**

# **📊 19: Highest stat requirement spells (end-game)**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_intelligence,
    required_faith,
    required_arcane,
    (required_intelligence + required_faith + required_arcane) AS total_reqs,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
ORDER BY total_reqs DESC
LIMIT 20;


spell_name spell_type required_intelligence required_faith required_arcane total_reqs fp_cost_min bonus Explosive Ghostflame Sorcery 42 30 0 72 29 Death Light of Miquella Incantation 0 72 0 72 48 Miquellan Rellana's Twin Moons Sorcery 72 0 0 72 47 Full Moon Rennala's Full Moon Sorcery 70 0 0 70 47 Full Moon Minor Erdtree Incantation 0 70 0 70 30 Erdtree Ranni's Dark Moon Sorcery 68 0 0 68 57 Full Moon Radagon's Rings of Light Incantation 31 31 0 62 21 Golden Order Comet Azur Sorcery 60 0 0 60 40 Primeval Sorceries Messmer's Orb Incantation 0 60 0 60 31 Messmer Fire Rykard's Rancor Sorcery 40 18 0 58 23 Magma Land of Shadow (Incantation) Incantation 0 58 0 58 40 Erdtree Ancient Death Rancor Sorcery 34 24 0 58 21 Death Vortex of Putrescence Sorcery 32 26 0 58 29 Death Meteorite of Astel Sorcery 55 0 0 55 60 Gravity Bayle's Flame Lightning Incantation 0 0 53 53 43 Dragon Communion Rain of Fire Incantation 0 52 0 52 27 Messmer Fire Comet Sorcery 52 0 0 52 24 Glintstone Founding Rain of Stars Sorcery 52 0 0 52 27 Primeval Sorceries Dragonbolt of Florissax Incantation 0 52 0 52 35 Dragon Cult Elden Stars Incantation 0 50 0 50 41 Erdtree

**The top 20 spells with the highest total stat requirements in Elden Ring include a mix of Incantations and Sorceries. Most of these spells demand either Faith or Intelligence, while a few require Arcane. For example, Light of Miquella and Minor Erdtree are Faith-heavy, while Rellana's Twin Moons and Explosive Ghostflame require high Intelligence. Their FP costs vary significantly, ranging from 21 to 60 FP, reflecting the balance between raw stat demands and resource expenditure. Bonuses like Erdtree, Full Moon, and Dragon Communion indicate thematic spell categories tied to specific in-game factions or mechanics.**

# **📊 20: Analyze stamina cost**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    stamina_cost,
    fp_cost_min
FROM eldenringcatalog.gold.fact_spell_usage
ORDER BY stamina_cost DESC
LIMIT 30;

spell_name spell_type stamina_cost fp_cost_min Greyoll's Roar Incantation 80.0 50 Placidusax's Ruin Incantation 75.0 62 Bayle's Tyranny Incantation 72.0 46 Dragonmaw Incantation 72.0 34 Smarag's Glintstone Breath Incantation 60.0 36 Light of Miquella Incantation 60.0 48 Agheel's Flame Incantation 60.0 36 Ekzykes's Decay Incantation 60.0 48 Ghostflame Breath Incantation 60.0 36 Theodorix's Magma Incantation 60.0 45 Borealis's Mist Incantation 60.0 48 Lansseax's Glaive Incantation 55.0 22 Fortissax's Lightning Spear Incantation 52.0 35 Law of Regression Incantation 50.0 55 Rotten Butterflies Incantation 50.0 48 Scarlet Aeonia Incantation 50.0 48 Protection of the Erdtree Incantation 50.0 30 Blessing of the Erdtree Incantation 50.0 60 Dragonclaw Incantation 50.0 24 Golden Vow Incantation 50.0 47 Erdtree Heal Incantation 50.0 65 Aspects of the Crucible: Horns Incantation 49.0 18 Dragonice Incantation 48.0 36 Land of Shadow (Incantation) Incantation 48.0 40 Rotten Breath Incantation 48.0 36 Elden Stars Incantation 48.0 41 Frozen Lightning Spear Incantation 45.0 29 Lord's Heal Incantation 45.0 55 Vortex of Putrescence Sorcery 45.0 29 Explosive Ghostflame Sorcery 45.0 29

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    stamina_cost,
    fp_cost_min,
    ROUND(stamina_cost / fp_cost_min, 2) AS stamina_fp_ratio
FROM eldenringcatalog.gold.fact_spell_usage
WHERE stamina_cost > 0 AND fp_cost_min > 0
ORDER BY stamina_fp_ratio DESC
LIMIT 20;

spell_name spell_type stamina_cost fp_cost_min stamina_fp_ratio Discus of Light Incantation 20.0 3 6.67 Briars of Sin Sorcery 26.0 6 4.33 Carian Slicer Sorcery 12.0 4 3.0 Glintstone Pebble Sorcery 20.0 7 2.86 Aspects of the Crucible: Horns Incantation 49.0 18 2.72 Bestial Sling Incantation 19.0 7 2.71 Beast Claw Incantation 27.0 10 2.7 Rejection Incantation 24.0 9 2.67 Bloodboon Incantation 34.0 13 2.62 Flame, Fall Upon Them Incantation 40.0 16 2.5 Lansseax's Glaive Incantation 55.0 22 2.5 Stone of Gurranq Incantation 37.0 15 2.47 Briars of Punishment Sorcery 22.0 9 2.44 Mantle of Thorns Sorcery 22.0 9 2.44 Honed Bolt Incantation 29.0 12 2.42 Gelmir's Fury Sorcery 38.0 16 2.38 Shatter Earth Sorcery 22.0 10 2.2 Great Glintstone Shard Sorcery 26.0 12 2.17 Glintstone Icecrag Sorcery 26.0 12 2.17 Dragonmaw Incantation 72.0 34 2.12

**The analysis of stamina-heavy spells shows that some incantations and sorceries consume a lot of stamina, with Greyoll's Roar (80), Placidusax's Ruin (75), and Bayle's Tyranny (72) topping the list. However, when considering efficiency, the stamina-to-FP ratio highlights spells that deliver more stamina effect per FP spent. For example, Discus of Light has an extremely high ratio of 6.67, followed by Briars of Sin at 4.33, making them highly efficient for stamina usage relative to FP cost. This ratio helps identify spells that are resource-efficient in combat while still being impactful in terms of stamina.**

# **📊 21: Best sorceries for pure INT builds**

In [0]:
%sql
SELECT 
    distinct spell_name,
    required_intelligence,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE spell_type = 'Sorcery'
    AND required_intelligence > 0
    AND required_faith = 0
    AND required_arcane = 0
ORDER BY required_intelligence DESC
LIMIT 20;

spell_name required_intelligence fp_cost_min bonus Rellana's Twin Moons 72 47 Full Moon Rennala's Full Moon 70 47 Full Moon Ranni's Dark Moon 68 57 Full Moon Comet Azur 60 40 Primeval Sorceries Meteorite of Astel 55 60 Gravity Comet 52 24 Glintstone Founding Rain of Stars 52 27 Primeval Sorceries Blades of Stone 48 18 Gravity Crystal Torrent 47 20 Crystalian Loretta's Mastery 46 39 Loretta's Sorceries Stars of Ruin 43 32 Primeval Sorceries Fleeting Microcosm 42 26 Finger Crystal Release 41 34 Crystalian Shattering Crystal 38 21 Crystalian Night Comet 38 21 Night Gravitational Missile 36 18 Gravity Collapsing Stars 36 18 Gravity Cherishing Fingers 36 20 Finger Glintstone Cometshard 36 17 Glintstone Zamor Ice Storm 36 17 Cold

**The top intelligence-based sorceries are dominated by high-tier spells requiring purely intelligence with no faith or arcane prerequisites. The most demanding spells are Rellana's Twin Moons (72 INT), Rennala's Full Moon (70 INT), and Ranni's Dark Moon (68 INT), all tied to the Full Moon bonus. Other notable entries include Comet Azur (60 INT, Primeval Sorceries) and Meteorite of Astel (55 INT, Gravity), showing that high intelligence unlocks some of the most potent sorceries in Elden Ring. These spells typically have higher FP costs, reflecting their power and utility in endgame scenarios.**

# **📊 22: Best incantations for pure FAI builds**

In [0]:
%sql
SELECT 
    distinct spell_name,
    required_faith,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE spell_type = 'Incantation'
    AND required_faith > 0
    AND required_intelligence = 0
    AND required_arcane = 0
ORDER BY required_faith DESC
LIMIT 20;

spell_name required_faith fp_cost_min bonus Light of Miquella 72 48 Miquellan Minor Erdtree 70 30 Erdtree Messmer's Orb 60 31 Messmer Fire Land of Shadow (Incantation) 58 40 Erdtree Dragonbolt of Florissax 52 35 Dragon Cult Rain of Fire 52 27 Messmer Fire Elden Stars 50 41 Erdtree Spira 48 10 Spiral Death Lightning 47 28 Dragon Cult Fortissax's Lightning Spear 46 35 Dragon Cult Black Blade 46 26 Erdtree Black Flame Ritual 42 30 Godskin Apostle Erdtree Heal 42 65 Erdtree Midra's Flame of Frenzy 41 22 Frenzied Flame Flame of the Fell God 41 34 Fire Giant Lansseax's Glaive 40 22 Dragon Cult Blessing of the Erdtree 38 60 Erdtree Knight's Lightning Spear 36 29 Dragon Cult Multilayered Ring of Light 36 23 Miquellan Placidusax's Ruin 36 62 -

**The most faith-intensive incantations in Elden Ring prioritize pure faith with no intelligence or arcane requirements. Leading the list is Light of Miquella (72 FAI, Miquellan) and Minor Erdtree (70 FAI, Erdtree), reflecting extremely powerful endgame healing or offensive spells. Other notable incantations include Messmer's Orb (60 FAI, Messmer Fire), Land of Shadow (58 FAI, Erdtree), and Dragonbolt of Florissax (52 FAI, Dragon Cult), which highlight the synergy between high faith and specific bonuses. These spells vary in FP cost, indicating a balance between raw power and resource consumption.**

# **📊 23: Hybrid INT/FAI spells**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_intelligence,
    required_faith,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE required_intelligence > 0 AND required_faith > 0
ORDER BY required_intelligence DESC, required_faith DESC;

spell_name spell_type required_intelligence required_faith fp_cost_min bonus Explosive Ghostflame Sorcery 42 30 29 Death Rykard's Rancor Sorcery 40 18 23 Magma Ancient Death Rancor Sorcery 34 24 21 Death Vortex of Putrescence Sorcery 32 26 29 Death Radagon's Rings of Light Incantation 31 31 21 Golden Order Mass of Putrescence Sorcery 28 22 41 Death Tibia's Summons Sorcery 28 20 17 Death Gelmir's Fury Sorcery 28 15 16 Magma Rings of Spectral Light Sorcery 24 18 14 Death Triple Rings of Light Incantation 23 23 23 Golden Order Fia's Mist Sorcery 23 18 25 Death Roiling Magma Sorcery 21 12 28 Magma Immutable Shield Incantation 19 19 15 Golden Order Magma Shot Sorcery 19 10 16 Magma Litany of Proper Death Incantation 17 17 17 Golden Order Rancorcall Sorcery 16 14 12 Death Order's Blade Incantation 13 13 22 Golden Order Discus of Light Incantation 13 13 3 Golden Order Order Healing Incantation 11 11 15 Golden Order

**These spells demand a hybrid investment in intelligence and faith, often combining powerful sorceries and incantations. Leading the pack is Explosive Ghostflame (42 INT, 30 FAI, Death bonus), followed by Rykard's Rancor (40 INT, 18 FAI, Magma) and Ancient Death Rancor (34 INT, 24 FAI, Death). Incantations like Radagon's Rings of Light (31 INT, 31 FAI, Golden Order) and Triple Rings of Light (23 INT, 23 FAI, Golden Order) highlight endgame versatility for hybrid builds. FP costs vary, but these spells generally offer strong effects in exchange for dual stat investment, ideal for characters balancing spellcasting and miracles.**

# **📊 24: Calculate "spell value" score**

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    fp_cost_min,
    (required_intelligence + required_faith + required_arcane) AS total_reqs,
    slots_required,
    ROUND(100 / ((fp_cost_min * 0.5) + ((required_intelligence + required_faith + required_arcane) * 0.3) + (slots_required * 10)), 2) AS value_score
FROM eldenringcatalog.gold.fact_spell_usage
WHERE fp_cost_min > 0 
    AND (required_intelligence + required_faith + required_arcane) > 0
ORDER BY value_score DESC
LIMIT 30;

spell_name spell_type fp_cost_min total_reqs slots_required value_score Surge, O Flame! Incantation 1 9 1 7.58 Swift Glintstone Shard Sorcery 5 12 1 6.21 Carian Slicer Sorcery 4 14 1 6.17 Cure Poison Incantation 7 10 1 6.06 Glintstone Pebble Sorcery 7 10 1 6.06 Bestial Sling Incantation 7 10 1 6.06 Catch Flame Incantation 10 8 1 5.75 Beast Claw Incantation 10 8 1 5.75 Bestial Constitution Incantation 10 9 1 5.65 Rejection Incantation 9 12 1 5.52 Glintstone Arc Sorcery 9 13 1 5.43 Flame Sling Incantation 11 10 1 5.41 Divine Bird Feathers Incantation 3 24 1 5.35 Thops's Barrier Sorcery 7 18 1 5.29 Night Shard Sorcery 7 18 1 5.29 Starlight Sorcery 9 15 1 5.26 Carian Retaliation Sorcery 8 17 1 5.24 Discus of Light Incantation 3 26 1 5.18 Shatter Earth Sorcery 10 15 1 5.13 Glintstone Stars Sorcery 12 12 1 5.10 Lord's Aid Incantation 12 12 1 5.10 Lucidity Sorcery 10 17 1 4.98 Unseen Blade Sorcery 13 12 1 4.98 Magic Glintblade Sorcery 12 14 1 4.95 Briars of Sin Sorcery 6 24 1 4.95 Fire Serpent Incantation 11 16 1 4.93 Urgent Heal Incantation 16 8 1 4.90 Glintstone Nail Sorcery 10 18 1 4.90 Glintstone Icecrag Sorcery 12 15 1 4.88 Assassin's Approach Incantation 15 10 1 4.88

**The highest-ranked spell is “Surge, O Flame!” (Incantation) with a minimal FP cost of 1, low total stat requirement of 9, and just 1 slot, yielding a value score of 7.58. Other top contenders include Swift Glintstone Shard, Carian Slicer, Cure Poison, Glintstone Pebble, and Bestial Sling, all combining low FP, moderate stat requirements, and efficient slot usage.**

**These spells represent high-efficiency choices for builds that want maximum utility with minimal resource investment, perfect for early-to-mid game or hybrid casters seeking strong performance without overcommitting stats or FP**

# **📊 25: Spells by school/category**

In [0]:
%sql
SELECT 
    bonus,
    COUNT(DISTINCT *) AS spell_count,
    ROUND(AVG(fp_cost_min), 2) AS avg_fp,
    ROUND(AVG(required_intelligence), 2) AS avg_int,
    ROUND(AVG(required_faith), 2) AS avg_fai,
    ROUND(AVG(required_arcane), 2) AS avg_arc
FROM eldenringcatalog.gold.fact_spell_usage
GROUP BY bonus
ORDER BY spell_count DESC;

bonus spell_count avg_fp avg_int avg_fai avg_arc Glintstone 18 16.78 20.56 0.0 0.0 Two Fingers 16 24.06 0.0 13.44 0.0 Dragon Communion 16 37.75 0.0 17.44 18.44 Dragon Cult 15 26.4 0.0 31.4 0.0 Erdtree 15 34.93 0.0 37.07 0.0 Carian 12 15.08 22.25 0.0 0.0 Golden Order 9 21.44 21.44 14.11 0.0 Fire Monk 8 16.13 0.0 13.38 0.0 Death 8 23.5 28.38 21.5 0.0 Night 7 17.0 22.29 0.0 0.0 Gravity 7 24.86 34.29 0.0 0.0 Bestial 7 14.0 0.0 11.57 0.0 Frenzied Flame 6 22.83 0.0 27.33 0.0 Servants of Rot 6 29.33 0.0 21.17 0.0 Godskin Apostle 5 22.6 0.0 27.0 0.0 Blood 5 15.4 0.0 13.8 17.0 Cold 5 18.2 23.8 0.0 0.0 Fire Giant 4 26.5 0.0 31.5 0.0 Magma 4 20.75 27.0 13.75 0.0 Finger 4 19.75 32.0 0.0 0.0 Spiral 4 14.5 0.0 32.5 0.0 Messmer Fire 3 23.0 0.0 42.67 0.0 Crystalian 3 25.0 42.0 0.0 0.0 Full Moon 3 50.33 70.0 0.0 0.0 Primeval Sorceries 3 33.0 51.67 0.0 0.0 Erdtree [1][2] 3 22.0 0.0 27.0 0.0 Miquellan 2 35.5 0.0 54.0 0.0 Loretta's Sorceries 2 31.5 36.0 0.0 0.0 Claymen Sorceries 2 14.0 22.0 0.0 16.5 - 2 44.0 0.0 27.5 0.0 Aberrant 2 7.5 0.0 22.5 0.0 Scadutree 2 12.0 0.0 22.0 0.0 -- 2 13.5 0.0 26.0 0.0 Glintstone (Stonedigger) 1 10.0 15.0 0.0 0.0 Godslayer 1 21.0 0.0 28.0 0.0 Glintstone (Stonediggers) 1 22.0 21.0 0.0 0.0

**The dataset shows that Glintstone spells are the most common, focusing on intelligence with low FP costs, while Erdtree and Dragon bonuses favor high faith and arcane, making them powerful but costly. Full Moon and Primeval Sorceries have extreme stat and FP requirements, reflecting their endgame strength. Niche bonuses like Claymen and Blood offer specialized utility for hybrid builds, balancing stats and FP for unique effects.**

# **📊 26: Spell recommendations by build**

## INTELLIGENCE BUILD (Best sorceries)

In [0]:
%sql
SELECT 
    distinct spell_name,
    required_intelligence,
    fp_cost_min,
    bonus,
    ROUND(required_intelligence / (fp_cost_min + 1), 2) AS efficiency
FROM eldenringcatalog.gold.fact_spell_usage
WHERE spell_type = 'Sorcery' AND required_intelligence > 0
ORDER BY efficiency DESC
LIMIT 10;

spell_name required_intelligence fp_cost_min bonus efficiency Carian Slicer 14 4 Carian 2.8 Miriam's Vanishing 26 9 Carian 2.6 Blades of Stone 48 18 Gravity 2.53 Night Shard 18 7 Night 2.25 Thops's Barrier 18 7 Glintstone 2.25 Crystal Torrent 47 20 Crystalian 2.24 Comet 52 24 Glintstone 2.08 Glintstone Cometshard 36 17 Glintstone 2.0 Swift Glintstone Shard 12 5 Glintstone 2.0 Zamor Ice Storm 36 17 Cold 2.0

## FAITH BUILD (Best incantations)

In [0]:
%sql
SELECT 
    distinct spell_name,
    required_faith,
    fp_cost_min,
    bonus,
    ROUND(required_faith / (fp_cost_min + 1), 2) AS efficiency
FROM eldenringcatalog.gold.fact_spell_usage
WHERE spell_type = 'Incantation' AND required_faith > 0
ORDER BY efficiency DESC
LIMIT 10;

spell_name required_faith fp_cost_min bonus efficiency Divine Bird Feathers 24 3 -- 6.0 Surge, O Flame! 9 1 Fire Monk 4.5 Spira 48 10 Spiral 4.36 Discus of Light 13 3 Golden Order 3.25 Minor Erdtree 70 30 Erdtree 2.26 Watchful Spirit 26 12 Spiral 2.0 Messmer's Orb 60 31 Messmer Fire 1.88 Rain of Fire 52 27 Messmer Fire 1.86 Honed Bolt 24 12 Dragon Cult 1.85 Aspects of the Crucible: Thorns 27 14 Erdtree 1.8

## ARCANE BUILD (Dragon/Blood spells)

In [0]:
%sql
SELECT 
    distinct spell_name,
    spell_type,
    required_arcane,
    fp_cost_min,
    bonus
FROM eldenringcatalog.gold.fact_spell_usage
WHERE required_arcane > 0
ORDER BY required_arcane DESC
LIMIT 10;

spell_name spell_type required_arcane fp_cost_min bonus Bayle's Flame Lightning Incantation 53 43 Dragon Communion Bayle's Tyranny Incantation 49 46 Dragon Communion Furious Blade of Ansbach Incantation 27 18 Blood Great Oracular Bubble Sorcery 18 16 Claymen Sorceries Bloodboon Incantation 17 13 Blood Greyoll's Roar Incantation 17 50 Dragon Communion Dragonmaw Incantation 16 34 Dragon Communion Swarm of Flies Incantation 16 14 Blood Bloodflame Talons Incantation 15 12 Blood Ghostflame Breath Incantation 15 36 Dragon Communion

**The efficiency analysis highlights spells that maximize their effect relative to FP cost for each primary stat. For Sorceries, spells like Carian Slicer and Miriam’s Vanishing deliver high intelligence scaling per FP, making them efficient choices for mage builds. In Incantations, Divine Bird Feathers and Surge, O Flame! stand out for faith-based efficiency, giving strong effects for minimal FP investment. For Arcane-focused spells, Bayle’s Flame Lightning and Bayle’s Tyranny dominate in raw arcane requirement, though their FP costs are higher, reflecting their specialized and powerful utility. Overall, this shows which spells provide the best value for investment in each stat category.**

## Key Findings
- Volume & Cost: Average FP cost ≈ 23.7 (range 1–65); high-cost spells are late‑game heals, legendary sorceries, dragon breaths.
- Memory Slots: Multi-slot spells are rare; only three require 3 slots (Placidusax’s Ruin, Scarlet Aeonia, Comet Azur).
- Stat Peaks:
  - Highest INT: Rellana’s Twin Moons (72), Rennala’s Full Moon (70), Ranni’s Dark Moon (68); sorcery INT avg ≈ 28.
  - Highest FAI: Light of Miquella (72), Minor Erdtree (70); incantation Faith avg ≈ 24.3.
  - Highest ARC: Bayle’s Flame Lightning (53), Bayle’s Tyranny (49); Arcane spells concentrated in Dragon/Blood groups.
- Hybrids: Explosive Ghostflame (42 INT / 30 FAI), Radagon’s Rings of Light (31 / 31), Triple Rings of Light (23 / 23) show dual-scaling niche power.
- FP Efficiency: Lowest FP-per-stat and top “value” spells include Surge, O Flame!, Swift Glintstone Shard, Carian Slicer, Glintstone Pebble, Cure Poison, Bestial Sling.
- Schools:
  - Glintstone: Most common, low FP, INT-focused.
  - Dragon / Erdtree: Fewer, higher FP and Faith/Arcane investment.
  - Full Moon / Primeval / Gravity: High stat + FP; end‑game burst/control.
  - Frenzied Flame: Mid Faith, aggressive utility; moderate FP.
- DLC vs Base: DLC spells slightly higher avg FP (≈24.8) and substantially higher Faith (≈24.7 vs 13.6); Base Game marginally higher avg INT.
- Stamina: Greyoll’s Roar (80), Placidusax’s Ruin (75) are stamina heavy; high stamina-to-FP efficiency examples include Discus of Light (ratio ~6.7) and Briars of Sin.
- Access & Onboarding: Beginner-friendly spells (low total stat 8–12) mostly small Faith incantations + early sorceries (Glintstone Pebble).

## Recommendations
### Intelligence (Pure Sorcery)
- Early core: Glintstone Pebble, Carian Slicer (cheap spam + melee reach).
- Mid-game: Rock Sling (gravity stagger), Swift Glintstone Shard (speed), Loretta’s spells (range).
- Late nukes: Comet Azur, Full Moon trilogy, Rellana’s Twin Moons (reserve 3-slot only if replacing other burst).
- Slot strategy: 1 high-impact (Full Moon / Comet Azur) + 2–3 low-FP sustain sorceries for dungeon clears.

### Faith (Pure Incantation)
- Early sustain: Catch Flame, Flame Sling, Heal / Urgent Heal, Cure Poison utility.
- Mid utility: Golden Vow, Blessing of the Erdtree (long buffs), Flame Grant Me Strength (hybrid AR boost).
- Late power: Light of Miquella, Minor Erdtree, Messmer’s Orb (situational boss usage).
- Keep one cheap offensive + one AoE + one buff; avoid stacking multiple long-cast heals in solo play.

### Arcane / Dragon / Blood
- Status application: Rotten Breath / Ekzykes’s Decay (rot), Swarm of Flies (bleed pressure).
- Debuff: Greyoll’s Roar (use sparingly—high stamina).
- High investment burst: Bayle’s Flame Lightning / Bayle’s Tyranny (consider flask allocation for FP sustain).
- Mix one rot/breath, one utility debuff, one direct damage lightning.

### Hybrid INT/FAI
- Versatile core: Radagon’s Rings of Light (fast holy beam), Triple Rings of Light (multi-lane control), Explosive Ghostflame (AoE DOT).
- Catalyst choice: Use Golden Order Seal or Prince of Death’s staff/seal pairing depending on scaling focus.
- Limit to 1–2 dual-stat high-cost spells; fill remaining slots with a cheap mono-stat tool for resource efficiency.

### Efficiency / Value Optimization
- Mob clear set: Surge, O Flame! (tick), Carian Slicer (melee), Swift Glintstone Shard / Glintstone Pebble (ranged).
- High value rotation: Alternate a low FP spell every other cast to preserve flasks in long dungeons.
- Track FP-per-stat; replace spells >0.6 ratio if a lower alternative covers similar role.

### Memory Slot Management
- Reserve 3-slot spells only if build centered on them (Comet Azur beam setups, Placidusax’s Ruin arena control).
- Typical optimal layout: 1 burst (large slot or high FP), 2–3 efficiency fillers, 1 buff, 1 utility/status.

### Stamina & Casting Flow
- Avoid chaining multiple high stamina spells before weapon attacks; stagger Greyoll’s Roar at opener then swap to low-stamina options.
- Use high stamina-to-FP ratio spells (Discus of Light, Briars of Sin) for sustained pressure in longer fights.

### Flask & Resource Strategy
- INT builds: Allocate slightly more FP flasks if running multiple high-cost nukes (e.g., Comet Azur + Full Moon).
- Faith builds: Balance HP vs FP; large heals reduce FP flask need—keep at least one low-FP emergency heal.
- Hybrid: Even flask split; rely on efficient value spells to prevent FP starvation.

### DLC Integration
- Faith builds gain stronger late-game scaling via DLC additions—justify pushing Faith >60 earlier.
- Reassess stat allocation: pure INT may add one hybrid beam (Radagon’s Rings) if leveling Faith marginally.

### Progression Path
1. Start: Secure cheap spam (Pebble / Catch Flame) + one utility (Heal / Cure Poison).
2. Mid: Introduce a control spell (Gravity / Rot) + buff (Golden Vow).
3. Late: Add signature nuke (Comet Azur / Light of Miquella) and optimize slot economy by pruning redundant mid-tier damage spells.

### Risk Mitigation
- High FP & multi-slot spells reduce flexibility—test encounter efficacy before permanent slot commitment.
- Preserve at least one rapid-cast, low commitment spell for interrupt windows and finishing blows.

## Quick Loadout Examples
- Pure INT (mob + boss): Carian Slicer | Swift Glintstone Shard | Rock Sling | Comet Azur | Rennala’s Full Moon | Golden Vow (if hybrid seal) | Utility (Cure Poison/Night Shard).
- Pure Faith: Catch Flame | Flame Sling | Golden Vow | Blessing of the Erdtree | Light of Miquella | Flame Grant Me Strength | Urgent Heal.
- Arcane/Status: Rotten Breath | Swarm of Flies | Greyoll’s Roar | Bayle’s Flame Lightning | Ekzykes’s Decay | Bestial Sling.
- Hybrid INT/FAI: Radagon’s Rings of Light | Triple Rings of Light | Explosive Ghostflame | Glintstone Pebble | Heal | Golden Vow.

## Prioritize
1. One low-FP spam.
2. One status/control (rot / gravity).
3. One burst nuke.
4. One buff.
5. One utility / heal.
6. Optional second nuke if slots allow.